glusterd2/commands/peers/deletepeer.go

```diff
@@ -43,13 +43,21 @@ func deletePeerHandler(w http.ResponseWriter, r *http.Request) {
 		restutils.SendHTTPError(ctx, w, status, err)
 		return
 	}
+
 	// You cannot remove yourself
 	if id == gdctx.MyUUID.String() {
 		logger.Debug("request denied, received request to delete self from cluster")
 		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "removing self is disallowed.")
 		return
 	}
 
+	//Check if peer liveness key is present in store
+	if _, alive := store.Store.IsNodeAlive(id); !alive {
+		logger.Error("can not delete peer, peer is not alive")
```

Comment:
can you add peer id while logging? it will help for debugging

Reply:
It's already there in logger fields